### ISSUE TITLE
Refresh the file icon within Explorer or another file browser

### ISSUE THREAD

**Issue by charltonstanley** [microsoft/vscode#74084]
I think it would be really nice if the icon next to the file name matched the current (or future? #71827) product icon. :)  This would make it easier to distinguish what program is going to open when i double click on this .ps1 file.

thanks!
![image](https://user-images.githubusercontent.com/22397984/58126006-891bdb00-7bd7-11e9-92b9-73d79a8afab2.png)

edit: grammer

**Comment by misolori** [comment]
This is actually from our system file icons (see #4845), the icon you are seeing is the default. We only support a small list of file icons and will keep this in mind.

**Comment by charltonstanley** [comment]
@misolori Looking through #4845, i understand that due to the sheer plethora of extensions available, that you can't create a custom icon for each one and have it open in vscode, I think it would still be reasonable to have the vscode icon be used when you set the default programs for specific file types. For example, MS Office Excel vs Libre Office. If I choose to open *.xlsx icons with Libre, the icon will look like the Excel version of Libre Office.  If I choose MS Office Excel, then it will look like the MS version.
To recap, I think vscode should be the same way.  When I set vscode as the default program for a file type, replace the icon for those file types.
Thank you!
